packages/amplify_core/lib/src/category/amplify_auth_category.dart

```diff
@@ -61,6 +125,67 @@ class AuthCategory extends AmplifyCategory<AuthPluginInterface> {
   /// {@template amplify_core.amplify_auth_category.confirm_sign_up}
   /// Confirm the current sign up for [username] with the [confirmationCode]
   /// provided by the user.
+  ///
+  /// <?code-excerpt "doc/lib/auth.dart" region="imports"?>
+  /// ```dart
+  /// import 'package:amplify_auth_cognito/amplify_auth_cognito.dart';
+  /// import 'package:amplify_flutter/amplify_flutter.dart';
+  /// ```
+  ///
+  /// <?code-excerpt "doc/lib/auth.dart" region="confirm-signup"?>
+  /// ```dart
+  /// Future<void> confirmUser({
+  ///   required String username,
+  ///   required String confirmationCode,
+  /// }) async {
+  ///   try {
+  ///     final result = await Amplify.Auth.confirmSignUp(
+  ///       username: username,
+  ///       confirmationCode: confirmationCode,
+  ///     );
+  ///     return _handleSignUpResult(result);
+  ///   } on AuthException catch (e) {
+  ///     safePrint('Error confirming user: ${e.message}');
+  ///   }
+  /// }
+  /// ```
+  ///
+  /// After confirming the sign up, check the [SignUpResult] object to check if
```

Comment:
Can we include the code description comments you have here directly in the auth docs?  I'm assuming that could make the `docs/lib/auth.dart` file more readable and better enforce separation on where we update the examples as current updates could feasibly be in `doc/lib/auth.dart` and `amplify_auth_category.dart`

Reply:
Unfortunately, no. The goal of this is to make maintaining code snippets easier, not necessarily to reduce the burden of maintaining docs in general.

Reply:
Alright, what are the downsides to consolidating the doc comments + code examples into the `doc/lib/auth.dart` file?  

From my understanding, there wouldn't be a case where the comment and the code are used separately, so if they're intertwined wouldn't it be good to keep them together?

Reply:
There isn't any downside, it's just not possible.

The code snippets work by injecting Dart code into the Markdown doc comments. The only way to colocate the Dart code and documentation is to affix it as documentation to the Dart code, e.g.

```dart
// #docregion handle-signup
/// After confirming the sign up, check the [SignUpResult] object to check if
/// further confirmations are needed or if the sign up is complete.
Future<void> _handleSignUpResult(SignUpResult result) async {
  switch (result.nextStep.signUpStep) {
    case AuthSignUpStep.confirmSignUp:
      final codeDeliveryDetails = result.nextStep.codeDeliveryDetails!;
      _handleCodeDelivery(codeDeliveryDetails);
      break;
    case AuthSignUpStep.done:
      safePrint('Sign up is complete');
      break;
  }
}
// #enddocregion handle-signup
```

But this has two issues.

1. This code is no longer reusable. `handleSignUpResult` is injected in multiple places in the Auth category's doc comments. This general comment about confirming sign up is not applicable to each instance where we want the code to appear.
2. In the rendered HTML documentation, this comment will no longer appear as paragraph text, but as a code comment which is not appropriate in this case.